aiohttp/client.py

```diff
@@ -326,7 +326,11 @@ def request(self, method, url, **kwargs):
                     tcp_nodelay(conn.transport, True)
                     tcp_cork(conn.transport, False)
                     try:
-                        resp = await req.send(conn)
+                        try:
+                            resp = await req.send(conn)
+                        except BaseException:
+                            conn.close()
+                            raise
```

Comment:
It seems we're don't the same thing twice. How about to regroup try/except a bit?
```
try:
  resp = await req.send(conn)
  try:
    await resp.start(conn, read_unit_eof)
  except:
    resp.close()
    raise
except:
  conn.close()
  raise
```

Reply:
Done.